src/cider/nrepl/middleware/test.clj

```diff
@@ -99,33 +99,67 @@
 ;; of type `:clojure.test.check.clojure-test/shrinking`, which `defspec`
 ;; produces to report failing input before shrinking it.
 
-(defn report
-  "Handle reporting for test events. This takes a test event map as an argument
-  and updates the `current-report` atom to reflect test results and summary
-  statistics."
+(defmulti report
+  "Handle reporting for test events.
+
+  This takes a test event map as an argument and updates the `current-report`
+  atom to reflect test results and summary statistics."
+  :type)
+
+(defmethod report :default [m])
```

Comment:
Seems to me it'd be a good idea to add some short docstrings to the methods, so people would get an idea what exactly are we handling with them.

Reply:
`defmethod`s don't accept docstrings, which is just as well--there's already a docstring for the `defmulti`, and the methods are just variants that dispatch on `:type`.

Reply:
Ah, I've forgotten this. Haven't used multimethods in quite a while. My point was mostly that some of the dispatch values might not mean much to casual readers of the code.